<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html>
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Add coverage to regions in <code><a href='http://www.rdocumentation.org/packages/GenomicRanges/topics/GRanges-class'>GRanges</a></code> object. — addCovToGR • sevenC</title>

<!-- jquery -->
<script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/1.7.1/clipboard.min.js" integrity="sha384-cV+rhyOuRHc9Ub/91rihWcGmMmCXDeksTtCihMupQHSsi8GIIRDG0ThDc3HGQFJ3" crossorigin="anonymous"></script>

<<<<<<< HEAD
<!-- sticky kit -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>



=======
<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../jquery.sticky-kit.min.js"></script>
<script src="../pkgdown.js"></script>
>>>>>>> cc85f59ba546b611db1046bd67d12e222efc59eb
<meta property="og:title" content="Add coverage to regions in <code><a href='http://www.rdocumentation.org/packages/GenomicRanges/topics/GRanges-class'>GRanges</a></code> object. — addCovToGR" />

<meta property="og:description" content="This function adds a NumericList  of coverage (or any
other signal in the input bigWig file) to each range in a
GRanges object. The coverage is reported for a
fixed-sized window around the region center. For regions with negative strand,
the coverage vector is reversed. The coverage signal is added as new metadata
column holding a NumericList object. Note, this
function does not work on windows because reading of bigWig files is currently
not supported on windows." />
<meta name="twitter:card" content="summary" />


<<<<<<< HEAD

=======
>>>>>>> cc85f59ba546b611db1046bd67d12e222efc59eb
<!-- mathjax -->
<script src='https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML'></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


<<<<<<< HEAD
=======


>>>>>>> cc85f59ba546b611db1046bd67d12e222efc59eb
  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">sevenC</a>
<<<<<<< HEAD
        <span class="label label-default" data-toggle="tooltip" data-placement="bottom" title="Released package">1.1.2</span>
=======
        <span class="label label-default" data-toggle="tooltip" data-placement="bottom" title="Released package">0.99.4</span>
>>>>>>> cc85f59ba546b611db1046bd67d12e222efc59eb
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/sevenC.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
      
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/ibn-salem/sevenC">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      
      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Add coverage to regions in <code><a href='http://www.rdocumentation.org/packages/GenomicRanges/topics/GRanges-class'>GRanges</a></code> object.</h1>
<<<<<<< HEAD
    <small class="dont-index">Source: <a href='https://github.com/ibn-salem/sevenC/blob/master/R/parseChIP-seq.R'><code>R/parseChIP-seq.R</code></a></small>
    <div class="hidden name"><code>addCovToGR.Rd</code></div>
    </div>

    <div class="ref-description">
=======
    
    <div class="hidden name"><code>addCovToGR.Rd</code></div>
    </div>

>>>>>>> cc85f59ba546b611db1046bd67d12e222efc59eb
    
    <p>This function adds a <code><a href='http://www.rdocumentation.org/packages/IRanges/topics/AtomicList-class'>NumericList</a></code>  of coverage (or any
other signal in the input bigWig file) to each range in a
<code><a href='http://www.rdocumentation.org/packages/GenomicRanges/topics/GRanges-class'>GRanges</a></code> object. The coverage is reported for a
fixed-sized window around the region center. For regions with negative strand,
the coverage vector is reversed. The coverage signal is added as new metadata
column holding a <code><a href='http://www.rdocumentation.org/packages/IRanges/topics/AtomicList-class'>NumericList</a></code> object. Note, this
function does not work on windows because reading of bigWig files is currently
not supported on windows.</p>
    
<<<<<<< HEAD
    </div>
=======
>>>>>>> cc85f59ba546b611db1046bd67d12e222efc59eb

    <pre class="usage"><span class='fu'>addCovToGR</span>(<span class='no'>gr</span>, <span class='no'>bwFile</span>, <span class='kw'>window</span> <span class='kw'>=</span> <span class='fl'>1000</span>, <span class='kw'>binSize</span> <span class='kw'>=</span> <span class='fl'>1</span>, <span class='kw'>colname</span> <span class='kw'>=</span> <span class='st'>"chip"</span>)</pre>
    
    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>gr</th>
      <td><p><code><a href='http://www.rdocumentation.org/packages/GenomicRanges/topics/GRanges-class'>GRanges</a></code> object with genomic regions. It
should contain a valid seqinfo object with defined seqlengths.</p></td>
    </tr>
    <tr>
      <th>bwFile</th>
      <td><p>File path or connection to BigWig or wig file with coverage to
parse from.</p></td>
    </tr>
    <tr>
      <th>window</th>
      <td><p>Numeric scalar for window size around the center of ranges in
<code>gr</code>.</p></td>
    </tr>
    <tr>
      <th>binSize</th>
      <td><p>Integer scalar as size of bins to which the coverage values are
combined.</p></td>
    </tr>
    <tr>
      <th>colname</th>
      <td><p>Character as name of the new column that is created in
<code>gr</code>.</p></td>
    </tr>
    </table>
    
    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p><code><a href='http://www.rdocumentation.org/packages/GenomicRanges/topics/GRanges-class'>GRanges</a></code> as input but with an additional
 meta column containing the coverage values for each region as
 <code><a href='http://www.rdocumentation.org/packages/IRanges/topics/AtomicList-class'>NumericList</a></code>.</p>
    

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='kw'>if</span> (<span class='no'>.Platform</span>$<span class='no'>OS.type</span> <span class='kw'>!=</span> <span class='st'>"windows"</span>) {

 <span class='co'># use example bigWig file of ChIP-seq signals on human chromosome 22</span>
 <span class='no'>exampleBigWig</span> <span class='kw'>&lt;-</span> <span class='fu'>system.file</span>(<span class='st'>"extdata"</span>,
 <span class='st'>"GM12878_Stat1.chr22_1-30000000.bigWig"</span>, <span class='kw'>package</span> <span class='kw'>=</span> <span class='st'>"sevenC"</span>)

 <span class='co'># use example CTCF moitf location on human chromosome 22</span>
 <span class='no'>motifGR</span> <span class='kw'>&lt;-</span> <span class='kw pkg'>sevenC</span><span class='kw ns'>::</span><span class='no'><a href='http://www.rdocumentation.org/packages/sevenC/topics/motif.hg19.CTCF.chr22'>motif.hg19.CTCF.chr22</a></span>

 <span class='co'># add ChIP-seq signals to motif regions</span>
 <span class='no'>motifGR</span> <span class='kw'>&lt;-</span> <span class='fu'>addCovToGR</span>(<span class='no'>motifGR</span>, <span class='no'>exampleBigWig</span>)

 <span class='co'># add ChIP-seq signals as column named "Stat1"</span>
 <span class='no'>motifGR</span> <span class='kw'>&lt;-</span> <span class='fu'>addCovToGR</span>(<span class='no'>motifGR</span>, <span class='no'>exampleBigWig</span>, <span class='kw'>colname</span> <span class='kw'>=</span> <span class='st'>"Stat1"</span>)

 <span class='co'># add ChIP-seq signals in windows of 500bp around motif centers</span>
 <span class='no'>motifGR</span> <span class='kw'>&lt;-</span> <span class='fu'>addCovToGR</span>(<span class='no'>motifGR</span>, <span class='no'>exampleBigWig</span>, <span class='kw'>window</span> <span class='kw'>=</span> <span class='fl'>500</span>)

 <span class='co'># add ChIP-seq signals in bins of 10 bp</span>
 <span class='no'>motifGR</span> <span class='kw'>&lt;-</span> <span class='fu'>addCovToGR</span>(<span class='no'>motifGR</span>, <span class='no'>exampleBigWig</span>, <span class='kw'>binSize</span> <span class='kw'>=</span> <span class='fl'>10</span>)

}</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#arguments">Arguments</a></li>
      
      <li><a href="#value">Value</a></li>
      
      <li><a href="#examples">Examples</a></li>
    </ul>

  </div>
</div>

      <footer>
      <div class="copyright">
  <p>Developed by Jonas Ibn-Salem.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
   </div>

  

  </body>
</html>

