<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html>
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Predict looping interactions. — predLoops • sevenC</title>

<!-- jquery -->
<script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/1.7.1/clipboard.min.js" integrity="sha384-cV+rhyOuRHc9Ub/91rihWcGmMmCXDeksTtCihMupQHSsi8GIIRDG0ThDc3HGQFJ3" crossorigin="anonymous"></script>

<!-- sticky kit -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>



<meta property="og:title" content="Predict looping interactions. — predLoops" />

<meta property="og:description" content="This function takes a GInteractions object with
candidate looping interactions. It should be annotated with features in
metadata columns. A logistic regression model is applied to predict looping
interaction probabilities." />
<meta name="twitter:card" content="summary" />



<!-- mathjax -->
<script src='https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML'></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">sevenC</a>
        <span class="label label-default" data-toggle="tooltip" data-placement="bottom" title="Released package">1.1.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/sevenC.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
      
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/ibn-salem/sevenC">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      
      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Predict looping interactions.</h1>
    <small class="dont-index">Source: <a href='https://github.com/ibn-salem/sevenC/blob/master/R/prediction.R'><code>R/prediction.R</code></a></small>
    <div class="hidden name"><code>predLoops.Rd</code></div>
    </div>

    <div class="ref-description">
    
    <p>This function takes a <code><a href='http://www.rdocumentation.org/packages/InteractionSet/topics/InteractionSet-class'>GInteractions</a></code> object with
candidate looping interactions. It should be annotated with features in
metadata columns. A logistic regression model is applied to predict looping
interaction probabilities.</p>
    
    </div>

    <pre class="usage"><span class='fu'>predLoops</span>(<span class='no'>gi</span>, <span class='kw'>formula</span> <span class='kw'>=</span> <span class='kw'>NULL</span>, <span class='kw'>betas</span> <span class='kw'>=</span> <span class='kw'>NULL</span>, <span class='kw'>colname</span> <span class='kw'>=</span> <span class='st'>"pred"</span>,
  <span class='kw'>cutoff</span> <span class='kw'>=</span> <span class='fu'>get</span>(<span class='st'>"cutoffBest10"</span>))</pre>
    
    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>gi</th>
      <td><p>A <code><a href='http://www.rdocumentation.org/packages/InteractionSet/topics/InteractionSet-class'>GInteractions</a></code> object with coverage
correlation and genomic features in metadata columns. See
<a href='prepareCisPairs.html'>prepareCisPairs</a> and <a href='addCor.html'>addCor</a> to build it.</p></td>
    </tr>
    <tr>
      <th>formula</th>
      <td><p>A <code><a href='http://www.rdocumentation.org/packages/stats/topics/formula'>formula</a></code>. All predictor variables should
be available in the in metadata columns of <code>gi</code>. If NULL, the following
default formula is used: <code>~ dist + strandOrientation + score_min +
chip</code>.</p></td>
    </tr>
    <tr>
      <th>betas</th>
      <td><p>A vector with parameter estimates for predictor variables. They
should be in the same order as variables in <code>formula</code>. Per default
estimates of <code><a href='modelBest10Avg.html'>modelBest10Avg</a></code> are used. See
<code><a href='modelBest10Avg.html'>?modelBest10Avg</a></code> for more detailed information on each parameter.</p></td>
    </tr>
    <tr>
      <th>colname</th>
      <td><p>A <code>character</code> as column name of new metadata column in
<code>gi</code> for predictions.</p></td>
    </tr>
    <tr>
      <th>cutoff</th>
      <td><p>Numeric cutoff on prediction score. Only interactions with
interaction probability &gt;= <code>cutoff</code> are reported. If <code>NULL</code>, all
input interactions are reported. Default is <code><a href='cutoffBest10.html'>cutoffBest10</a></code>, an
optimal cutoff based on F1-score on 10 best performing transcription factor
ChIP-seq data sets. See <code><a href='cutoffBest10.html'>?cutoffBest10</a></code> for more details.</p></td>
    </tr>
    </table>
    
    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>A <code><a href='http://www.rdocumentation.org/packages/InteractionSet/topics/InteractionSet-class'>GInteractions</a></code> as <code>gi</code> with an
 additional metadata column holding the predicted looping probability.</p>
    
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'><p><code><a href='prepareCisPairs.html'>prepareCisPairs</a></code>, <code><a href='addCor.html'>addCor</a></code></p></div>
    

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'>
<span class='co'># use example CTCF moitf location on human chromosome 22 with chip coverage</span>
<span class='no'>motifGR</span> <span class='kw'>&lt;-</span> <span class='kw pkg'>sevenC</span><span class='kw ns'>::</span><span class='no'><a href='http://www.rdocumentation.org/packages/sevenC/topics/motif.hg19.CTCF.chr22.cov'>motif.hg19.CTCF.chr22.cov</a></span>

<span class='co'># build candidate interactions</span>
<span class='no'>gi</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='prepareCisPairs.html'>prepareCisPairs</a></span>(<span class='no'>motifGR</span>)

<span class='co'># add ChIP-seq signals correlation</span>
<span class='no'>gi</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='addCovCor.html'>addCovCor</a></span>(<span class='no'>gi</span>)

<span class='co'># predict chromatin looping interactions</span>
<span class='no'>loops</span> <span class='kw'>&lt;-</span> <span class='fu'>predLoops</span>(<span class='no'>gi</span>)

<span class='co'># add prediction score for all candidates without filter</span>
<span class='no'>gi</span> <span class='kw'>&lt;-</span> <span class='fu'>predLoops</span>(<span class='no'>gi</span>, <span class='kw'>cutof</span> <span class='kw'>=</span> <span class='kw'>NULL</span>)

<span class='co'># add prediction score using custom column name</span>
<span class='no'>gi</span> <span class='kw'>&lt;-</span> <span class='fu'>predLoops</span>(<span class='no'>gi</span>, <span class='kw'>cutof</span> <span class='kw'>=</span> <span class='kw'>NULL</span>, <span class='kw'>colname</span> <span class='kw'>=</span> <span class='st'>"my_colname"</span>)

<span class='co'># Filter loop predictions on custom cutoff</span>
<span class='no'>loops</span> <span class='kw'>&lt;-</span> <span class='fu'>predLoops</span>(<span class='no'>gi</span>, <span class='kw'>cutoff</span> <span class='kw'>=</span> <span class='fl'>0.4</span>)

<span class='co'># predict chromatin looping interactions using custom model parameters</span>
<span class='no'>myParams</span> <span class='kw'>&lt;-</span> <span class='fu'>c</span>(-<span class='fl'>4</span>, -<span class='fl'>5</span>, -<span class='fl'>2</span>, -<span class='fl'>1</span>, -<span class='fl'>1</span>, <span class='fl'>5</span>, <span class='fl'>3</span>)
<span class='no'>loops</span> <span class='kw'>&lt;-</span> <span class='fu'>predLoops</span>(<span class='no'>gi</span>, <span class='kw'>betas</span> <span class='kw'>=</span> <span class='no'>myParams</span>)

<span class='co'># predict chromatin loops using custom model formula and params</span>
<span class='no'>myFormula</span> <span class='kw'>&lt;-</span> ~ <span class='no'>dist</span> + <span class='no'>score_min</span>
<span class='co'># define parameters for intercept, dist and motif_min</span>
<span class='no'>myParams</span> <span class='kw'>&lt;-</span> <span class='fu'>c</span>(-<span class='fl'>5</span>, -<span class='fl'>4</span>, <span class='fl'>6</span>)
<span class='no'>loops</span> <span class='kw'>&lt;-</span> <span class='fu'>predLoops</span>(<span class='no'>gi</span>, <span class='kw'>formula</span> <span class='kw'>=</span> <span class='no'>myFormula</span>, <span class='kw'>betas</span> <span class='kw'>=</span> <span class='no'>myParams</span>)</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#arguments">Arguments</a></li>
      
      <li><a href="#value">Value</a></li>

      <li><a href="#see-also">See also</a></li>
      
      <li><a href="#examples">Examples</a></li>
    </ul>

  </div>
</div>

      <footer>
      <div class="copyright">
  <p>Developed by Jonas Ibn-Salem.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
   </div>

  

  </body>
</html>

