<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Prediction of chromatin looping interactions with <em>sevenC</em> • sevenC</title>
<!-- jquery --><script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script><!-- Font Awesome icons --><link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/1.7.1/clipboard.min.js" integrity="sha384-cV+rhyOuRHc9Ub/91rihWcGmMmCXDeksTtCihMupQHSsi8GIIRDG0ThDc3HGQFJ3" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Prediction of chromatin looping interactions with &lt;em&gt;sevenC&lt;/em&gt;">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">sevenC</a>
        <span class="label label-default" data-toggle="tooltip" data-placement="bottom" title="Released package">1.1.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/sevenC.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/ibn-salem/sevenC">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Prediction of chromatin looping interactions with <em>sevenC</em>
</h1>
                        <h4 class="author">Jonas Ibn-Salem</h4>
            <address class="author_afil">
      Faculty of Biology, Johannes Gutenberg University of Mainz, 55128 Mainz, Germany<br><a class="author_email" href="mailto:#"></a><a href="mailto:j.ibn-salem@uni-mainz.de">j.ibn-salem@uni-mainz.de</a>
      </address>
                              <h4 class="author">Miguel Andrade-Navarro</h4>
            <address class="author_afil">
      Faculty of Biology, Johannes Gutenberg University of Mainz, 55128 Mainz, Germany<br><small class="dont-index">Source: <a href="https://github.com/ibn-salem/sevenC/blob/master/vignettes/sevenC.Rmd"><code>vignettes/sevenC.Rmd</code></a></small>
      <div class="hidden name"><code>sevenC.Rmd</code></div>

    </address>
</div>

    
        <div class="abstract">
      <p class="abstract">Abstract</p>
      Chromatin looping is an essential feature of eukaryotic genomes and can bring regulatory sequences, such as enhancers or transcription factor binding sites, in the close physical proximity of regulated target genes. Here, we provide sevenC, an R package that uses protein binding signals from ChIP-seq and sequence motif information to predict chromatin looping events. Cross-linking of proteins that bind close to loop anchors result in ChIP-seq signals at both anchor loci. These signals are used at CTCF motif pairs together with their distance and orientation to each other to predict whether they interact or not. The resulting chromatin loops might be used to associate enhancers or transcription factor binding sites (e.g., ChIP-seq peaks) to regulated target genes.
    </div>
    
<div id="background-and-introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#background-and-introduction" class="anchor"></a>Background and introduction</h1>
<p>Gene expression is regulated by binding of transcription factors (TF) to genomic DNA. However, many binding sites are in distal regulatory regions, such as enhancers, that are hundreds of kilobases apart from genes. These regulatory regions can physically interact with promoters of regulated genes by chromatin looping interactions. These looping interaction can be measured genome-wide by chromatin conformation capture techniques such as Hi-C or ChIA-PET <span class="citation">(Rao et al. 2014; Tang et al. 2015)</span>. Despite many exciting insights into the three-dimensional organization of genomes, these experimental methods are not only elaborate and expansive but also have limited resolution and are only available for a limited number of cell types and conditions. In contrast, the binding sites of TFs can be detected genome-wide by ChIP-seq experiment with high resolution and are available for hundreds of TFs in many cell type and conditions. However, classical analysis of ChIP-seq gives only the direct binding sites of targeted TFs (ChIP-seq peaks) and it is not trivial to associate them to the regulated gene without chromatin looping information. Therefore, we provide a computational method to predict chromatin interactions from only genomic sequence features and ChIP-seq data. The predicted looping interactions can be used to associate TF binding sites (ChIP-seq peaks) or enhancers to regulated genes and thereby improve functional downstream analysis on the level of genes.</p>
<p>In this vignette, we show how to use the R package <em><a href="http://bioconductor.org/packages/sevenC">sevenC</a></em> to predict chromatin looping interactions between CTCF motifs by using only ChIP-seq data form a single experiment. Furthermore, we show how to train the prediction model using custom data.</p>
<p>A more detailed explanation of the sevenC method together with prediction performance analysis is available in the associated preprint <span class="citation">(Ibn-Salem and Andrade-Navarro 2018)</span>.</p>
</div>
<div id="installation" class="section level1">
<h1 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h1>
<p>To install the <em>sevenC</em> package, start R and enter:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="co"># install.packages("BiocManager")</span></a>
<a class="sourceLine" id="cb1-2" data-line-number="2">BiocManager<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/BiocManager/topics/install">install</a></span>(<span class="st">"sevenC"</span>)</a></code></pre></div>
</div>
<div id="predict-chromatin-looping-interactions" class="section level1">
<h1 class="hasAnchor">
<a href="#predict-chromatin-looping-interactions" class="anchor"></a>Predict chromatin looping interactions</h1>
<div id="basic-usage-example" class="section level2">
<h2 class="hasAnchor">
<a href="#basic-usage-example" class="anchor"></a>Basic usage example</h2>
<p>Here we show how to use the <em><a href="http://bioconductor.org/packages/sevenC">sevenC</a></em> package with default options to predict chromatin looping interactions among CTCF motif locations on the human chromosome 22. As input, we only use CTCF motif locations and a single bigWig file from a STAT1 ChIP-seq experiment in human GM12878 cells <span class="citation">(Dunham et al. 2012)</span>.</p>
<div id="get-motif-pairs" class="section level3">
<h3 class="hasAnchor">
<a href="#get-motif-pairs" class="anchor"></a>Get motif pairs</h3>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1"><span class="kw">library</span>(sevenC)</a>
<a class="sourceLine" id="cb2-2" data-line-number="2"></a>
<a class="sourceLine" id="cb2-3" data-line-number="3"><span class="co"># load provided CTCF motifs in human genome</span></a>
<a class="sourceLine" id="cb2-4" data-line-number="4">motifs &lt;-<span class="st"> </span>motif.hg19.CTCF.chr22</a>
<a class="sourceLine" id="cb2-5" data-line-number="5"></a>
<a class="sourceLine" id="cb2-6" data-line-number="6"><span class="co"># get motifs pairs</span></a>
<a class="sourceLine" id="cb2-7" data-line-number="7">gi &lt;-<span class="st"> </span><span class="kw"><a href="../reference/prepareCisPairs.html">prepareCisPairs</a></span>(motifs)</a></code></pre></div>
</div>
<div id="add-chip-seq-data-and-compute-correaltion" class="section level3">
<h3 class="hasAnchor">
<a href="#add-chip-seq-data-and-compute-correaltion" class="anchor"></a>Add ChIP-seq data and compute correaltion</h3>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1"><span class="co"># use example ChIP-seq bigWig file</span></a>
<a class="sourceLine" id="cb3-2" data-line-number="2">bigWigFile &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">"extdata"</span>, <span class="st">"GM12878_Stat1.chr22_1-30000000.bigWig"</span>, </a>
<a class="sourceLine" id="cb3-3" data-line-number="3">  <span class="dt">package =</span> <span class="st">"sevenC"</span>)</a>
<a class="sourceLine" id="cb3-4" data-line-number="4"></a>
<a class="sourceLine" id="cb3-5" data-line-number="5"><span class="co"># add ChIP-seq coverage and compute correaltion at motif pairs</span></a>
<a class="sourceLine" id="cb3-6" data-line-number="6">gi &lt;-<span class="st"> </span><span class="kw"><a href="../reference/addCor.html">addCor</a></span>(gi, bigWigFile)</a></code></pre></div>
</div>
<div id="predict-loops" class="section level3">
<h3 class="hasAnchor">
<a href="#predict-loops" class="anchor"></a>Predict loops</h3>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1"><span class="co"># predict looping interactions among all motif pairs</span></a>
<a class="sourceLine" id="cb4-2" data-line-number="2">loops &lt;-<span class="st"> </span><span class="kw"><a href="../reference/predLoops.html">predLoops</a></span>(gi)</a></code></pre></div>
</div>
</div>
<div id="more-detailed-usage-example" class="section level2">
<h2 class="hasAnchor">
<a href="#more-detailed-usage-example" class="anchor"></a>More detailed usage example</h2>
<p>Here we show in more detail each step of the loop prediction process. Again, we want to predict chromatin looping interactions among CTCF motif locations on chromosome 22 using a ChIP-seq for STAT1 in human GM12878 cells.</p>
<div id="prepare-ctcf-motif-pairs" class="section level3">
<h3 class="hasAnchor">
<a href="#prepare-ctcf-motif-pairs" class="anchor"></a>Prepare CTCF motif pairs</h3>
<p>First, we need to prepare CTCF motif pairs as candidate anchors for chromatin loop interactions. We use CTCF motif hits in human chromosome 22 as provide by <em><a href="http://bioconductor.org/packages/sevenC">sevenC</a></em> package. In general, any CTCF motifs can be used if provided as <code>GRanges</code>. To use the motif similarity score as a predictive feature, the motif data should contain -log<sub>10</sub> transformed p-values describing the significance of each motif hit. Here, we use CTCF motif sites as provided from the JASPAR genome browser tracks <span class="citation">(Khan et al. 2018)</span>. The objedt <code>motif.hg19.CTCF.chr22</code> in the <code>r BiocStyle::Biocpkg("sevenC")</code> package contains CTCF motif locations on chromosome 22. For more information on the motif data set, see <code><a href="../reference/motif.hg19.CTCF.html">?motif.hg19.CTCF</a></code>.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1"><span class="kw">library</span>(sevenC)</a>
<a class="sourceLine" id="cb5-2" data-line-number="2"></a>
<a class="sourceLine" id="cb5-3" data-line-number="3"><span class="co"># load provided CTCF motifs</span></a>
<a class="sourceLine" id="cb5-4" data-line-number="4">motifs &lt;-<span class="st"> </span>motif.hg19.CTCF.chr22</a></code></pre></div>
<p>The CTCF motif are represented as <code>GRanges</code> object from the <code>r BiocStyle::Biocpkg("GenomicRanges")</code> package. There are 917 CTCF motif locations on chromosome 22. The genome assembly is hg19. one metadata column named <code>score</code> shows motif match similarity as -log<sub>10</sub> transformed p-value.</p>
</div>
<div id="add-chip-seq-signals-at-motifs-sites" class="section level3">
<h3 class="hasAnchor">
<a href="#add-chip-seq-signals-at-motifs-sites" class="anchor"></a>Add ChIP-seq signals at motifs sites</h3>
<p>To predict loops, we need the ChIP-seq signals at all motif sites. Therefore, we read an example bigWig file with ChIP-seq signals.</p>
<p>An example file with only data on a subset of chromosome 22 is provided as part of the <em><a href="http://bioconductor.org/packages/sevenC">sevenC</a></em> package. The full file can be downloaded from ENCODE <span class="citation">(Dunham et al. 2012)</span> <a href="http://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeSydhTfbs/wgEncodeSydhTfbsGm12878Stat1StdSig.bigWig">here</a>. The file contains for each position in the genome the log-fold-change of ChIP-seq signals versus input control.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" data-line-number="1"><span class="co"># use example ChIP-seq bigWig file</span></a>
<a class="sourceLine" id="cb6-2" data-line-number="2">bigWigFile &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">"extdata"</span>, <span class="st">"GM12878_Stat1.chr22_1-30000000.bigWig"</span>, </a>
<a class="sourceLine" id="cb6-3" data-line-number="3">  <span class="dt">package =</span> <span class="st">"sevenC"</span>)</a></code></pre></div>
<p>We add ChIP-seq signals to all motifs in a window of 1000 bp using the function <code><a href="../reference/addCovToGR.html">addCovToGR()</a></code> as follows.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" data-line-number="1"><span class="co"># read ChIP-seq coverage </span></a>
<a class="sourceLine" id="cb7-2" data-line-number="2">motifs &lt;-<span class="st"> </span><span class="kw"><a href="../reference/addCovToGR.html">addCovToGR</a></span>(motifs, bigWigFile)</a></code></pre></div>
<p>This adds a new metadata column to <code>motifs</code> holding a <code>NumericList</code> with ChIP-seq signals for each motif location.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb8-1" data-line-number="1">motifs<span class="op">$</span>chip</a></code></pre></div>
<pre><code>## NumericList of length 917
## [["chr22"]] 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 ... 1 1 1 1 1 1 1 1 1 1 1 1 1 1
## [["chr22"]] 1 1 1 1 1 0 0 0 0 0 0 0 0 0 0 ... 2 2 2 2 2 2 2 2 2 2 2 2 2 2
## [["chr22"]] 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 ... 0 0 0 0 0 0 0 0 0 0 0 0 0 0
## [["chr22"]] 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 ... 1 1 1 1 1 1 1 1 1 1 1 1 1 1
## [["chr22"]] 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 ... 0 0 0 0 0 0 0 0 0 0 0 0 0 0
## [["chr22"]] 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 ... 0 0 0 0 0 0 0 0 0 0 0 0 0 0
## [["chr22"]] 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 ... 0 0 0 0 0 0 0 0 0 0 0 0 0 0
## [["chr22"]] 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 ... 0 0 0 0 0 0 0 0 0 0 0 0 0 0
## [["chr22"]] 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 ... 0 0 0 0 0 0 0 0 0 0 0 0 0 0
## [["chr22"]] 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 ... 2 2 2 2 2 2 2 2 2 2 2 2 2 2
## ...
## &lt;907 more elements&gt;</code></pre>
<p>Please note, on Windows systems, reading of bigWig files is currently not supported. See <code>help(rtracklayer::import.bw)</code> for more information. Users on Windows need to get ChIP-seq signals around motif sites as a <code>NumierList</code> object. A <code>NumericList</code> <code>l</code> with ChIP-signal counts around each motif list can be added by <code>motifs$chip &lt;- l</code>.</p>
</div>
<div id="build-pairs-of-motifs-as-candidate-interactions" class="section level3">
<h3 class="hasAnchor">
<a href="#build-pairs-of-motifs-as-candidate-interactions" class="anchor"></a>Build pairs of motifs as candidate interactions</h3>
<p>Now we build a dataset with all pairs of CTCF motif within 1 Mb and annotate it with distance, motif orientation, and motif score.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb10-1" data-line-number="1">gi &lt;-<span class="st"> </span><span class="kw"><a href="../reference/prepareCisPairs.html">prepareCisPairs</a></span>(motifs, <span class="dt">maxDist =</span> <span class="dv">10</span><span class="op">^</span><span class="dv">6</span>)</a>
<a class="sourceLine" id="cb10-2" data-line-number="2"></a>
<a class="sourceLine" id="cb10-3" data-line-number="3">gi</a></code></pre></div>
<pre><code>## StrictGInteractions object with 26076 interactions and 5 metadata columns:
##           seqnames1           ranges1     seqnames2           ranges2 |
##               &lt;Rle&gt;         &lt;IRanges&gt;         &lt;Rle&gt;         &lt;IRanges&gt; |
##       [1]     chr22 16186188-16186206 ---     chr22 16205307-16205325 |
##       [2]     chr22 16186188-16186206 ---     chr22 16238548-16238566 |
##       [3]     chr22 16186188-16186206 ---     chr22 16239188-16239206 |
##       [4]     chr22 16186188-16186206 ---     chr22 16239827-16239845 |
##       [5]     chr22 16186188-16186206 ---     chr22 16247246-16247264 |
##       ...       ...               ... ...       ...               ... .
##   [26072]     chr22 51110780-51110798 ---     chr22 51160028-51160046 |
##   [26073]     chr22 51110780-51110798 ---     chr22 51172084-51172102 |
##   [26074]     chr22 51130130-51130148 ---     chr22 51160028-51160046 |
##   [26075]     chr22 51130130-51130148 ---     chr22 51172084-51172102 |
##   [26076]     chr22 51160028-51160046 ---     chr22 51172084-51172102 |
##                dist strandOrientation   score_1   score_2 score_min
##           &lt;integer&gt;       &lt;character&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;
##       [1]     19119           forward      6.05      5.72      5.72
##       [2]     52360           forward      6.05      5.92      5.92
##       [3]     53000           forward      6.05       5.9       5.9
##       [4]     53639           forward      6.05      5.92      5.92
##       [5]     61058           forward      6.05      5.92      5.92
##       ...       ...               ...       ...       ...       ...
##   [26072]     49248           reverse      6.21      5.64      5.64
##   [26073]     61304           reverse      6.21       5.9       5.9
##   [26074]     29898           reverse      6.33      5.64      5.64
##   [26075]     41954           reverse      6.33       5.9       5.9
##   [26076]     12056           reverse      5.64       5.9      5.64
##   -------
##   regions: 917 ranges and 2 metadata columns
##   seqinfo: 93 sequences (1 circular) from hg19 genome</code></pre>
<p>The function <code><a href="../reference/prepareCisPairs.html">prepareCisPairs()</a></code> returns a <code>GInteractoin</code> object from the <code>r BiocStyle::Biocpkg("InteractonSet")</code> package, representing all motif pairs within the defined distance. The metadata columns of the <code>GInteractoin</code> object hold the genomic distance between motifs in bp (<code>dist</code>), the orientation of motifs (<code>strandOrientation</code>), and the motif score as -log<sub>10</sub> of the motif hit p-value (<code>score_1</code>, <code>score_2</code>, and <code>score_min</code>). Note, that the function <code><a href="../reference/prepareCisPairs.html">prepareCisPairs()</a></code> is a wrapper for three individual functions that perform each step separately and allow more options. First, <code><a href="../reference/getCisPairs.html">getCisPairs()</a></code> is used to builds the <code>GInteractoin</code> object. Than <code><a href="../reference/addStrandCombination.html">addStrandCombination()</a></code> adds the four possible strand combinations of motifs pairs. Finally, <code><a href="../reference/addMotifScore.html">addMotifScore()</a></code> adds the minimum motif score for each pair. These genomic features are used later as predictive variables.</p>
</div>
</div>
<div id="compute-chip-seq-similarity-at-motif-pairs" class="section level2">
<h2 class="hasAnchor">
<a href="#compute-chip-seq-similarity-at-motif-pairs" class="anchor"></a>Compute ChIP-seq similarity at motif pairs</h2>
<p>Now, we compute the similarity of ChIP-seq signals for all motif pairs as the correlation of signals across positions around motif centers. Thereby, for two motifs the corresponding ChIP-seq signal vectors that were added to <code>motifs</code> before, are compared by Pearson correlation. A high correlation of ChIP-seq signals at two motifs indicates a similar ChIP-seq coverage profile at the two motifs. This, in turn, is characteristic for physical interaction via chromatin looping, where ChIP signals are found on both sides with a similar distance to motif centers <span class="citation">(Ibn-Salem and Andrade-Navarro 2018)</span>. The correlation coefficient is added as additional metadata column to <code>gi</code>.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb12-1" data-line-number="1"><span class="co"># add ChIP-seq coverage and compute correaltion at motif pairs</span></a>
<a class="sourceLine" id="cb12-2" data-line-number="2">gi &lt;-<span class="st"> </span><span class="kw"><a href="../reference/addCovCor.html">addCovCor</a></span>(gi)</a></code></pre></div>
</div>
<div id="predict-loops-1" class="section level2">
<h2 class="hasAnchor">
<a href="#predict-loops-1" class="anchor"></a>Predict loops</h2>
<p>Now we can predict chromatin loops integrating from the ChIP-seq correlation and other genomic features in a logistic regression model. This is implemented in the <code><a href="../reference/predLoops.html">predLoops()</a></code> function.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb13-1" data-line-number="1">loops &lt;-<span class="st"> </span><span class="kw"><a href="../reference/predLoops.html">predLoops</a></span>(gi)</a>
<a class="sourceLine" id="cb13-2" data-line-number="2"></a>
<a class="sourceLine" id="cb13-3" data-line-number="3">loops</a></code></pre></div>
<pre><code>## StrictGInteractions object with 607 interactions and 7 metadata columns:
##         seqnames1           ranges1     seqnames2           ranges2 |
##             &lt;Rle&gt;         &lt;IRanges&gt;         &lt;Rle&gt;         &lt;IRanges&gt; |
##     [1]     chr22 16186188-16186206 ---     chr22 16239188-16239206 |
##     [2]     chr22 16205307-16205325 ---     chr22 16409786-16409804 |
##     [3]     chr22 16238548-16238566 ---     chr22 16409786-16409804 |
##     [4]     chr22 17398482-17398500 ---     chr22 17539262-17539280 |
##     [5]     chr22 17398482-17398500 ---     chr22 17652850-17652868 |
##     ...       ...               ... ...       ...               ... .
##   [603]     chr22 26756542-26756560 ---     chr22 27041422-27041440 |
##   [604]     chr22 26763796-26763814 ---     chr22 27041422-27041440 |
##   [605]     chr22 28123236-28123254 ---     chr22 28221328-28221346 |
##   [606]     chr22 29225543-29225561 ---     chr22 29436847-29436865 |
##   [607]     chr22 29436847-29436865 ---     chr22 29641698-29641716 |
##              dist strandOrientation   score_1   score_2 score_min
##         &lt;integer&gt;       &lt;character&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;
##     [1]     53000           forward      6.05       5.9       5.9
##     [2]    204479        convergent      5.72      5.79      5.72
##     [3]    171238        convergent      5.92      5.79      5.79
##     [4]    140780        convergent      6.48      6.89      6.48
##     [5]    254368        convergent      6.48      8.49      6.48
##     ...       ...               ...       ...       ...       ...
##   [603]    284880           reverse      8.27      7.93      7.93
##   [604]    277626           reverse      8.59      7.93      7.93
##   [605]     98092           reverse      6.29      8.16      6.29
##   [606]    211304           reverse      6.73      6.33      6.33
##   [607]    204851           reverse      6.33      7.27      6.33
##                  cor_chip              pred
##                 &lt;numeric&gt;         &lt;numeric&gt;
##     [1] 0.885072700379391 0.193088582740316
##     [2] 0.415730555504532  0.16264353243483
##     [3]             0.675 0.301511861680806
##     [4] 0.790283850445726 0.550518859830263
##     [5] 0.221485126152438 0.172598431306202
##     ...               ...               ...
##   [603] 0.349559060111433 0.225007521722854
##   [604] 0.739379609297975 0.423643299218372
##   [605] 0.601548182884723 0.188784650506129
##   [606] 0.928965197273641 0.245978189154254
##   [607] 0.909619042677005 0.242532301825788
##   -------
##   regions: 917 ranges and 2 metadata columns
##   seqinfo: 93 sequences (1 circular) from hg19 genome</code></pre>
<p>The <code><a href="../reference/predLoops.html">predLoops()</a></code> function returns a subset of motif pairs that are predicted to interact. The interactions are annotated with ChIP-seq correlation in column <code>cor_chip</code>. The column <code>pred</code> holds the predicted interaction probability according to the logistic regression model.</p>
<p>Note, that without specifying further options, the function <code><a href="../reference/predLoops.html">predLoops()</a></code> uses a default model that was optimized for several transcription factor ChIP-seq datasets by using experimental chromatin loops from Hi-C and ChIA-PET for validations <span class="citation">(Ibn-Salem and Andrade-Navarro 2018)</span>. However, users can specify custom features using the <code>formula</code> argument and provide custom parameters using the <code>betas</code> argument. Furthermore, per default the <code><a href="../reference/predLoops.html">predLoops()</a></code> function report only looping interactions that reach a minimal prediction score threshold. The fraction of reported loops can be modified using the <code>cutoff</code> argument.</p>
</div>
</div>
<div id="downstream-analysis-with-predicted-chromatin-loops" class="section level1">
<h1 class="hasAnchor">
<a href="#downstream-analysis-with-predicted-chromatin-loops" class="anchor"></a>Downstream analysis with predicted chromatin loops</h1>
<div id="linking-sets-of-regions" class="section level2">
<h2 class="hasAnchor">
<a href="#linking-sets-of-regions" class="anchor"></a>Linking sets of regions</h2>
<p>Predicted loops are represented as <code>GInteraction</code> and can, therefore, be used easily for downstream analysis with functions from the <code>r BiocStyle::Biocpkg("InteractonSet")</code> package. For example, linking two sets of regions (like ChIP-seq peaks and genes) can be done using the <code>linkOverlaps</code> function. See the <a href="http://bioconductor.org/packages/release/bioc/vignettes/InteractionSet/inst/doc/interactions.html">vignette</a> from the <em><a href="http://bioconductor.org/packages/InteractonSet">InteractonSet</a></em> package for more details and examples on working with <code>GInteraction</code> objects.</p>
</div>
<div id="write-predicted-loops-to-an-output-file" class="section level2">
<h2 class="hasAnchor">
<a href="#write-predicted-loops-to-an-output-file" class="anchor"></a>Write predicted loops to an output file</h2>
<p>Since looping interactions are stored as <code>GInteraction</code> objects, they can be exported as <a href="http://bedtools.readthedocs.io/en/latest/content/general-usage.html#bedpe-format">BEDPE</a> files using functions from <em><a href="http://bioconductor.org/packages/GenomicInteractions">GenomicInteractions</a></em> package. These files can be used for visualization in genome browsers or the <a href="https://www.aidenlab.org/juicebox/">Juicebox</a> tool.</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb15-1" data-line-number="1"><span class="kw">library</span>(GenomicInteractions)</a>
<a class="sourceLine" id="cb15-2" data-line-number="2"></a>
<a class="sourceLine" id="cb15-3" data-line-number="3"><span class="co"># export to output file</span></a>
<a class="sourceLine" id="cb15-4" data-line-number="4"><span class="kw"><a href="http://www.rdocumentation.org/packages/GenomicInteractions/topics/export.bedpe">export.bedpe</a></span>(loops, <span class="st">"loop_interactions.bedpe"</span>, <span class="dt">score =</span> <span class="st">"pred"</span>)</a></code></pre></div>
</div>
</div>
<div id="train-prediction-model-using-custom-data" class="section level1">
<h1 class="hasAnchor">
<a href="#train-prediction-model-using-custom-data" class="anchor"></a>Train prediction model using custom data</h1>
<p>Here, we show how to use <em><a href="http://bioconductor.org/packages/sevenC">sevenC</a></em> to build and train a logistic regression model for loop prediction.</p>
<div id="prepare-motif-pairs-and-add-chip-seq-data" class="section level2">
<h2 class="hasAnchor">
<a href="#prepare-motif-pairs-and-add-chip-seq-data" class="anchor"></a>Prepare motif pairs and add ChIP-seq data</h2>
<p>First, we need to build the pairs of motifs as candidates and add the ChIP-seq data as shown above.</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb16-1" data-line-number="1"><span class="co"># load provided CTCF motifs</span></a>
<a class="sourceLine" id="cb16-2" data-line-number="2">motifs &lt;-<span class="st"> </span>motif.hg19.CTCF.chr22</a>
<a class="sourceLine" id="cb16-3" data-line-number="3"></a>
<a class="sourceLine" id="cb16-4" data-line-number="4"><span class="co"># use example ChIP-seq coverage file</span></a>
<a class="sourceLine" id="cb16-5" data-line-number="5">bigWigFile &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">"extdata"</span>, <span class="st">"GM12878_Stat1.chr22_1-30000000.bigWig"</span>, </a>
<a class="sourceLine" id="cb16-6" data-line-number="6">  <span class="dt">package =</span> <span class="st">"sevenC"</span>)</a>
<a class="sourceLine" id="cb16-7" data-line-number="7"></a>
<a class="sourceLine" id="cb16-8" data-line-number="8"><span class="co"># add ChIP-seq coverage</span></a>
<a class="sourceLine" id="cb16-9" data-line-number="9">motifs &lt;-<span class="st"> </span><span class="kw"><a href="../reference/addCovToGR.html">addCovToGR</a></span>(motifs, bigWigFile)</a>
<a class="sourceLine" id="cb16-10" data-line-number="10"></a>
<a class="sourceLine" id="cb16-11" data-line-number="11"><span class="co"># build motif pairs</span></a>
<a class="sourceLine" id="cb16-12" data-line-number="12">gi &lt;-<span class="st"> </span><span class="kw"><a href="../reference/prepareCisPairs.html">prepareCisPairs</a></span>(motifs, <span class="dt">maxDist =</span> <span class="dv">10</span><span class="op">^</span><span class="dv">6</span>)</a>
<a class="sourceLine" id="cb16-13" data-line-number="13"></a>
<a class="sourceLine" id="cb16-14" data-line-number="14"><span class="co"># add correaltion of ChIP-signal</span></a>
<a class="sourceLine" id="cb16-15" data-line-number="15">gi &lt;-<span class="st"> </span><span class="kw"><a href="../reference/addCovCor.html">addCovCor</a></span>(gi)</a></code></pre></div>
</div>
<div id="train-predictor-with-known-loops" class="section level2">
<h2 class="hasAnchor">
<a href="#train-predictor-with-known-loops" class="anchor"></a>Train predictor with known loops</h2>
<p>We need to label true looping interactions by using experimental data of chromatin interactions. Here, we use loops from high-resolution Hi-C experiments in human GM12878 cells <span class="citation">(Rao et al. 2014)</span>. An example file with loops on chromosome 22 is provided with the <em><a href="http://bioconductor.org/packages/sevenC">sevenC</a></em> package and the function <code><a href="../reference/parseLoopsRao.html">parseLoopsRao()</a></code> reads loops in the format provided by Rao et al. and returns a <code>GInteraction</code> object.</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb17-1" data-line-number="1"><span class="co"># parse known loops</span></a>
<a class="sourceLine" id="cb17-2" data-line-number="2">knownLoopFile &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">"extdata"</span>, </a>
<a class="sourceLine" id="cb17-3" data-line-number="3">  <span class="st">"GM12878_HiCCUPS.chr22_1-30000000.loop.txt"</span>, <span class="dt">package =</span> <span class="st">"sevenC"</span>)</a>
<a class="sourceLine" id="cb17-4" data-line-number="4"></a>
<a class="sourceLine" id="cb17-5" data-line-number="5">knownLoops &lt;-<span class="st"> </span><span class="kw"><a href="../reference/parseLoopsRao.html">parseLoopsRao</a></span>(knownLoopFile)</a></code></pre></div>
<p>We can add a new metadata column to the motif pairs <code>gi</code>, indicating whether the pair is interacting in the experimental data using the function <code><a href="../reference/addInteractionSupport.html">addInteractionSupport()</a></code>.</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb18-1" data-line-number="1"><span class="co"># add known loops</span></a>
<a class="sourceLine" id="cb18-2" data-line-number="2">gi &lt;-<span class="st"> </span><span class="kw"><a href="../reference/addInteractionSupport.html">addInteractionSupport</a></span>(gi, knownLoops)</a></code></pre></div>
<p>The experimental support is added as factor with levels <code>"Loop"</code> and <code>"No loop"</code> as metadata column named <code>loop</code>. The column name can be modified using the <code>colname</code> argument.</p>
</div>
<div id="train-logistic-regression-model" class="section level2">
<h2 class="hasAnchor">
<a href="#train-logistic-regression-model" class="anchor"></a>Train logistic regression model</h2>
<p>We can use the R function <code>glm()</code> to fit a logistic regression model in which the <code>loop</code> column is the dependent variable and the ChIP-seq correlation, distance, and strand orientation are the predictors.</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb19-1" data-line-number="1">fit &lt;-<span class="st"> </span><span class="kw">glm</span>(</a>
<a class="sourceLine" id="cb19-2" data-line-number="2">  <span class="dt">formula =</span> loop <span class="op">~</span><span class="st"> </span>cor_chip <span class="op">+</span><span class="st"> </span>dist <span class="op">+</span><span class="st"> </span>strandOrientation, </a>
<a class="sourceLine" id="cb19-3" data-line-number="3">  <span class="dt">data =</span> <span class="kw">mcols</span>(gi), </a>
<a class="sourceLine" id="cb19-4" data-line-number="4">  <span class="dt">family =</span> <span class="kw">binomial</span>()</a>
<a class="sourceLine" id="cb19-5" data-line-number="5">  )</a></code></pre></div>
</div>
<div id="predict-loops-with-a-custom-model" class="section level2">
<h2 class="hasAnchor">
<a href="#predict-loops-with-a-custom-model" class="anchor"></a>Predict loops with a custom model</h2>
<p>Now, we can use this model to add predicted looping probabilities.</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb20-1" data-line-number="1"><span class="co"># add predict loops</span></a>
<a class="sourceLine" id="cb20-2" data-line-number="2">gi &lt;-<span class="st"> </span><span class="kw"><a href="../reference/predLoops.html">predLoops</a></span>(</a>
<a class="sourceLine" id="cb20-3" data-line-number="3">  gi,</a>
<a class="sourceLine" id="cb20-4" data-line-number="4">  <span class="dt">formula =</span> loop <span class="op">~</span><span class="st"> </span>cor_chip <span class="op">+</span><span class="st"> </span>dist <span class="op">+</span><span class="st"> </span>strandOrientation,</a>
<a class="sourceLine" id="cb20-5" data-line-number="5">  <span class="dt">betas =</span> <span class="kw">coef</span>(fit),</a>
<a class="sourceLine" id="cb20-6" data-line-number="6">  <span class="dt">cutoff =</span> <span class="ot">NULL</span></a>
<a class="sourceLine" id="cb20-7" data-line-number="7">)</a></code></pre></div>
<p>Here, we have to use the same formula as argument as in the model fitting step above. The <code>betas</code> argument takes the coefficients of the logistic regression model. Finally, the argument <code>cutoff = NULL</code> ensures that no filtering is done and all input candidates are reported. The prediction score is added as a new metadata column to <code>gi</code>.</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb21-1" data-line-number="1">gi </a></code></pre></div>
<pre><code>## StrictGInteractions object with 26076 interactions and 8 metadata columns:
##           seqnames1           ranges1     seqnames2           ranges2 |
##               &lt;Rle&gt;         &lt;IRanges&gt;         &lt;Rle&gt;         &lt;IRanges&gt; |
##       [1]     chr22 16186188-16186206 ---     chr22 16205307-16205325 |
##       [2]     chr22 16186188-16186206 ---     chr22 16238548-16238566 |
##       [3]     chr22 16186188-16186206 ---     chr22 16239188-16239206 |
##       [4]     chr22 16186188-16186206 ---     chr22 16239827-16239845 |
##       [5]     chr22 16186188-16186206 ---     chr22 16247246-16247264 |
##       ...       ...               ... ...       ...               ... .
##   [26072]     chr22 51110780-51110798 ---     chr22 51160028-51160046 |
##   [26073]     chr22 51110780-51110798 ---     chr22 51172084-51172102 |
##   [26074]     chr22 51130130-51130148 ---     chr22 51160028-51160046 |
##   [26075]     chr22 51130130-51130148 ---     chr22 51172084-51172102 |
##   [26076]     chr22 51160028-51160046 ---     chr22 51172084-51172102 |
##                dist strandOrientation   score_1   score_2 score_min
##           &lt;integer&gt;       &lt;character&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;
##       [1]     19119           forward      6.05      5.72      5.72
##       [2]     52360           forward      6.05      5.92      5.92
##       [3]     53000           forward      6.05       5.9       5.9
##       [4]     53639           forward      6.05      5.92      5.92
##       [5]     61058           forward      6.05      5.92      5.92
##       ...       ...               ...       ...       ...       ...
##   [26072]     49248           reverse      6.21      5.64      5.64
##   [26073]     61304           reverse      6.21       5.9       5.9
##   [26074]     29898           reverse      6.33      5.64      5.64
##   [26075]     41954           reverse      6.33       5.9       5.9
##   [26076]     12056           reverse      5.64       5.9      5.64
##                     cor_chip     loop                 pred
##                    &lt;numeric&gt; &lt;factor&gt;            &lt;numeric&gt;
##       [1]  0.339540495724593  No loop  0.00207736832933179
##       [2] -0.104691613939358  No loop 0.000422511191326569
##       [3]  0.885072700379391  No loop   0.0122895154866494
##       [4] -0.170960689666582  No loop 0.000335909138644671
##       [5] -0.104691613939358  No loop 0.000414237393073598
##       ...                ...      ...                  ...
##   [26072]               &lt;NA&gt;  No loop                 &lt;NA&gt;
##   [26073]               &lt;NA&gt;  No loop                 &lt;NA&gt;
##   [26074]               &lt;NA&gt;  No loop                 &lt;NA&gt;
##   [26075]               &lt;NA&gt;  No loop                 &lt;NA&gt;
##   [26076]               &lt;NA&gt;  No loop                 &lt;NA&gt;
##   -------
##   regions: 917 ranges and 2 metadata columns
##   seqinfo: 93 sequences (1 circular) from hg19 genome</code></pre>
<p>As a very simple validation, we can now compare the prediction score for looping and non-looping motif pairs using a boxplot.</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb23-1" data-line-number="1"><span class="kw">boxplot</span>(gi<span class="op">$</span>pred <span class="op">~</span><span class="st"> </span>gi<span class="op">$</span>loop, </a>
<a class="sourceLine" id="cb23-2" data-line-number="2">        <span class="dt">ylab =</span> <span class="st">"Predicted interaction probability"</span>)</a></code></pre></div>
<p><img src="sevenC_files/figure-html/unnamed-chunk-22-1.png" width="288"></p>
<p>The plot shows higher prediction scores for truly looping motif pairs. However, this is an insufficient evaluation of prediction performance, since the prediction score is evaluated on the same data as it was trained. A more detailed evaluation of prediction performance using cross-validation and different cell types is described in the 7C paper <span class="citation">(Ibn-Salem and Andrade-Navarro 2018)</span>.</p>
</div>
<div id="session-info" class="section level2">
<h2 class="hasAnchor">
<a href="#session-info" class="anchor"></a>Session info</h2>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb24-1" data-line-number="1"><span class="kw">sessionInfo</span>()</a></code></pre></div>
<pre><code>## R version 3.5.1 (2018-07-02)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: Ubuntu 16.04.5 LTS
## 
## Matrix products: default
## BLAS: /home/ibnsalem/bin/R-3.5.1/lib64/R/lib/libRblas.so
## LAPACK: /home/ibnsalem/bin/R-3.5.1/lib64/R/lib/libRlapack.so
## 
## locale:
##  [1] LC_CTYPE=de_DE.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=de_DE.UTF-8        LC_COLLATE=de_DE.UTF-8    
##  [5] LC_MONETARY=de_DE.UTF-8    LC_MESSAGES=de_DE.UTF-8   
##  [7] LC_PAPER=de_DE.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=de_DE.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] parallel  stats4    stats     graphics  grDevices utils     datasets 
## [8] methods   base     
## 
## other attached packages:
##  [1] GenomicInteractions_1.14.0  sevenC_1.0.0               
##  [3] InteractionSet_1.8.0        SummarizedExperiment_1.10.1
##  [5] DelayedArray_0.6.5          BiocParallel_1.14.2        
##  [7] matrixStats_0.54.0          Biobase_2.40.0             
##  [9] GenomicRanges_1.32.6        GenomeInfoDb_1.16.0        
## [11] IRanges_2.14.11             S4Vectors_0.18.3           
## [13] BiocGenerics_0.26.0         BiocStyle_2.8.2            
## 
## loaded via a namespace (and not attached):
##  [1] colorspace_1.3-2         rprojroot_1.3-2         
##  [3] biovizBase_1.28.2        htmlTable_1.12          
##  [5] XVector_0.20.0           base64enc_0.1-3         
##  [7] fs_1.2.6                 dichromat_2.0-0         
##  [9] rstudioapi_0.7           roxygen2_6.1.0          
## [11] bit64_0.9-7              AnnotationDbi_1.42.1    
## [13] xml2_1.2.0               splines_3.5.1           
## [15] knitr_1.20               Formula_1.2-3           
## [17] Rsamtools_1.32.3         cluster_2.0.7-1         
## [19] readr_1.1.1              compiler_3.5.1          
## [21] httr_1.3.1               backports_1.1.2         
## [23] assertthat_0.2.0         Matrix_1.2-14           
## [25] lazyeval_0.2.1           acepack_1.4.1           
## [27] htmltools_0.3.6          prettyunits_1.0.2       
## [29] tools_3.5.1              bindrcpp_0.2.2          
## [31] igraph_1.2.2             gtable_0.2.0            
## [33] glue_1.3.0               GenomeInfoDbData_1.1.0  
## [35] dplyr_0.7.6              Rcpp_0.12.18            
## [37] pkgdown_1.1.0            Biostrings_2.48.0       
## [39] rtracklayer_1.40.6       xfun_0.3                
## [41] stringr_1.3.1            ensembldb_2.4.1         
## [43] XML_3.98-1.16            zlibbioc_1.26.0         
## [45] MASS_7.3-50              scales_1.0.0            
## [47] BSgenome_1.48.0          VariantAnnotation_1.26.1
## [49] hms_0.4.2                ProtGenerics_1.12.0     
## [51] AnnotationFilter_1.4.0   RColorBrewer_1.1-2      
## [53] yaml_2.2.0               curl_3.2                
## [55] memoise_1.1.0            gridExtra_2.3           
## [57] ggplot2_3.0.0            biomaRt_2.36.1          
## [59] rpart_4.1-13             latticeExtra_0.6-28     
## [61] stringi_1.2.4            RSQLite_2.1.1           
## [63] desc_1.2.0               checkmate_1.8.5         
## [65] GenomicFeatures_1.32.2   boot_1.3-20             
## [67] rlang_0.2.2              pkgconfig_2.0.2         
## [69] commonmark_1.5           bitops_1.0-6            
## [71] evaluate_0.11            lattice_0.20-35         
## [73] purrr_0.2.5              bindr_0.1.1             
## [75] GenomicAlignments_1.16.0 htmlwidgets_1.2         
## [77] bit_1.1-14               tidyselect_0.2.4        
## [79] plyr_1.8.4               magrittr_1.5            
## [81] bookdown_0.7             R6_2.2.2                
## [83] Hmisc_4.1-1              DBI_1.0.0               
## [85] pillar_1.3.0             foreign_0.8-71          
## [87] survival_2.42-6          RCurl_1.95-4.11         
## [89] nnet_7.3-12              tibble_1.4.2            
## [91] crayon_1.3.4             rmarkdown_1.10          
## [93] progress_1.2.0           grid_3.5.1              
## [95] data.table_1.11.4        blob_1.1.1              
## [97] digest_0.6.16            munsell_0.5.0           
## [99] Gviz_1.24.0</code></pre>
</div>
</div>
<div id="references" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<div id="refs" class="references">
<div id="ref-Dunham2012">
<p>Dunham, Ian, Anshul Kundaje, Shelley F Aldred, Patrick J Collins, Carrie a Davis, Francis Doyle, Charles B Epstein, et al. 2012. “An integrated encyclopedia of DNA elements in the human genome.” <em>Nature</em> 489 (7414):57–74. <a href="https://doi.org/10.1038/nature11247" class="uri">https://doi.org/10.1038/nature11247</a>.</p>
</div>
<div id="ref-Ibn-Salem2018">
<p>Ibn-Salem, Jonas, and Miguel A. Andrade-Navarro. 2018. “Computational Chromosome Conformation Capture by Correlation of ChIP-seq at CTCF motifs.” <em>bioRxiv</em>, February. Cold Spring Harbor Laboratory, 257584. <a href="https://doi.org/10.1101/257584" class="uri">https://doi.org/10.1101/257584</a>.</p>
</div>
<div id="ref-Khan2018">
<p>Khan, Aziz, Oriol Fornes, Arnaud Stigliani, Marius Gheorghe, Jaime A Castro-Mondragon, Robin van der Lee, Adrien Bessy, et al. 2018. “JASPAR 2018: update of the open-access database of transcription factor binding profiles and its web framework.” <em>Nucleic Acids Research</em> 46 (D1):D260–D266. <a href="https://doi.org/10.1093/nar/gkx1126" class="uri">https://doi.org/10.1093/nar/gkx1126</a>.</p>
</div>
<div id="ref-Rao2014">
<p>Rao, Suhas S P, Miriam H Huntley, Neva C Durand, Elena K Stamenova, Ivan D Bochkov, James T Robinson, Adrian L Sanborn, et al. 2014. “A 3D map of the human genome at kilobase resolution reveals principles of chromatin looping.” <em>Cell</em> 159 (7). Elsevier Inc.:1665–80. <a href="https://doi.org/10.1016/j.cell.2014.11.021" class="uri">https://doi.org/10.1016/j.cell.2014.11.021</a>.</p>
</div>
<div id="ref-Tang2015">
<p>Tang, Zhonghui, Oscar Junhong Luo, Xingwang Li, Meizhen Zheng, Jacqueline Jufen Zhu, Przemyslaw Szalaj, Pawel Trzaskoma, et al. 2015. “CTCF-Mediated Human 3D Genome Architecture Reveals Chromatin Topology for Transcription.” <em>Cell</em>. Elsevier Inc., 1–17. <a href="https://doi.org/10.1016/j.cell.2015.11.024" class="uri">https://doi.org/10.1016/j.cell.2015.11.024</a>.</p>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#background-and-introduction">Background and introduction</a></li>
      <li><a href="#installation">Installation</a></li>
      <li>
<a href="#predict-chromatin-looping-interactions">Predict chromatin looping interactions</a><ul class="nav nav-pills nav-stacked">
<li><a href="#basic-usage-example">Basic usage example</a></li>
      <li><a href="#more-detailed-usage-example">More detailed usage example</a></li>
      <li><a href="#compute-chip-seq-similarity-at-motif-pairs">Compute ChIP-seq similarity at motif pairs</a></li>
      <li><a href="#predict-loops-1">Predict loops</a></li>
      </ul>
</li>
      <li>
<a href="#downstream-analysis-with-predicted-chromatin-loops">Downstream analysis with predicted chromatin loops</a><ul class="nav nav-pills nav-stacked">
<li><a href="#linking-sets-of-regions">Linking sets of regions</a></li>
      <li><a href="#write-predicted-loops-to-an-output-file">Write predicted loops to an output file</a></li>
      </ul>
</li>
      <li>
<a href="#train-prediction-model-using-custom-data">Train prediction model using custom data</a><ul class="nav nav-pills nav-stacked">
<li><a href="#prepare-motif-pairs-and-add-chip-seq-data">Prepare motif pairs and add ChIP-seq data</a></li>
      <li><a href="#train-predictor-with-known-loops">Train predictor with known loops</a></li>
      <li><a href="#train-logistic-regression-model">Train logistic regression model</a></li>
      <li><a href="#predict-loops-with-a-custom-model">Predict loops with a custom model</a></li>
      <li><a href="#session-info">Session info</a></li>
      </ul>
</li>
      <li><a href="#references">References</a></li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Jonas Ibn-Salem.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
</div>

  

  </body>
</html>
